packaging/tags.py

```diff
@@ -420,16 +431,141 @@ def _have_compatible_glibc(required_major, minimum_minor):
     return _check_glibc_version(version_str, required_major, minimum_minor)
 
 
+# Python does not provide platform information at sufficient granularity to
+# identify the architecture of the running executable in some cases, so we
+# determine it dynamically by reading the information from the running
+# process. This only applies on Linux, which uses the ELF format.
+class _ELFFileHeader(object):
+    # https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#File_header
+    class _InvalidELFFileHeader(ValueError):
+        """
+        An invalid ELF file header was found.
+        """
+
+    ELF_MAGIC_NUMBER = 0x7F454C46
+    ELFCLASS32 = 1
+    ELFCLASS64 = 2
+    ELFDATA2LSB = 1
+    ELFDATA2MSB = 2
+    EM_386 = 0x03
+    EM_ARM = 0x28
+    EF_ARM_ABIMASK = 0xFF000000
+    EF_ARM_ABI_VER5 = 0x05000000
+    EF_ARM_ABI_FLOAT_HARD = 0x00000400
+
+    def __init__(self, file):
+        # type: (_ELFFileHeader, IO[bytes]) -> None
```

Comment:
The type of `self` is typically omitted (see: [`Tag`](https://github.com/pypa/packaging/blob/8f7f708cdc78a0109e73bdb60c0e07055bee4307/packaging/tags.py#L59) above).

Reply:
done